lib/pure/asyncdispatch.nim

```diff
@@ -133,13 +153,43 @@
 ## however that this does not wait for completion, and you should use
 ## ``waitFor`` for that purpose.
 ##
-## Examples
-## ========
-##
 ## For examples take a look at the documentation for the modules implementing
 ## asynchronous IO. A good place to start is the
 ## `asyncnet module <asyncnet.html>`_.
 ##
+##
+## waitFor
+## -------
+##
+## ``waitFor`` blocks and runs the async loop until the Future finishes.
+##
+## Example:
+##
+## .. code-block:: nim
+##   proc example() {.async.} =
+##     echo "waitFor Example"
+##
+##   waitFor example()
+##
+## await
+## -----
+##
+## ``await`` will wait for the ``Future``, and gives you the contents inside the ``Future``,
+## ``await`` only available inside a function with the ``{.async.}`` or ``{.multisync.}`` Pragmas.
+##
+## Comparison
+## ----------
+##
+## ================================ ============ ========= =======
+##  Description                      asyncCheck   waitFor   await
+## ================================ ============ ========= =======
+## Waits for the Future to complete  No           Yes       Yes
+## Ignores the Future                Yes          No        No
+## Returns result inside Future      No           No        Yes
```

Comment:
"Returns result inside Future"  waitFor should be yes?

Example:

```nim
import asyncdispatch

proc foo(): Future[int] {.async.} =
  return 42

echo waitFor foo()
```

Reply:
This also fixes https://github.com/nim-lang/Nim/issues/5021